static/inventory.js

```diff
@@ -116,59 +118,94 @@ function showSettings() {
 }
 
 // Construct HTML of the results. String concatenation was chosen for rendering speed.
-var displayItems = function(items) {
+var displayItems = function(items, byType = false) {
     var resultDiv = $("#results");
     resultDiv.empty();
     displayId++;
-    displayItemsAsync(items, 0, resultDiv, displayId);
+    if (byType) displayItemsByTypeAsync(items, 0, resultDiv, displayId);
+    else displayItemsAsync(items, 0, resultDiv, displayId);
+};
+
+function displayItemsByTypeAsync(items, start, div, id, currentItemType = null) {
```

Comment:
You probably don't need to pass currentItemType. You can just do 

`var currentItemType = items[start].type`

no ?

Reply:
Indeed! It's simpler.